### DISCUSSION TITLE
generate_projects.sh script behaves differently on different devices

### DISCUSSION THREAD

**Discussion by farbod1277** [NVIDIA-Omniverse/PhysX#112] [Q&A] [answered]
Hi, I'm trying to automate a pipeline with docker and build PhysX within it and I noticed when running ./generate_projects.sh that to build for aarch64 or clang (the two options it presents) are in different order on different devices. For example (0) is clang on the pipeline that I'm trying to automate but (0) is aarch64 on my laptop. This makes it difficult to automate the pipeline by simply being able to pass in a (0) or (1) into the script based on your hardware because (0) and (1) are different in different environments.

![image](https://user-images.githubusercontent.com/20006748/223904427-6e525b33-bc0d-45ff-90b6-e23d33e00b3a.png)

vs

![image](https://user-images.githubusercontent.com/20006748/223904549-102dcaa0-d865-46cf-bd84-2975e442499e.png)

**Comment by preist-nvidia** [comment]
Maybe @ayoub-belarbi knows a good workaround for scripting here.

**Comment by ayoub-belarbi** [comment, marked as answer]
Hello @farbod1277,

There is a simple solution. You can pass the compiler name as an argument to that script, e.g.:
.`/generate_projects.sh linux` or `./generate_projects.sh linux-aarch64`.
Basically the preset name you can add as an argument.

Thanks

**Reply by farbod1277** [reply]
Thank you!
